adapters/colossus/colossus.go

```diff
@@ -29,21 +28,25 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters
 func (a *ColossusAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 	var errs []error
 	var err error
-	var tagID string
 
+	var bidderExt adapters.ExtImpBidder
+	var collossusExt openrtb_ext.ExtImpColossus
 	var adapterRequests []*adapters.RequestData
 
 	reqCopy := *request
 	for _, imp := range request.Imp {
 		reqCopy.Imp = []openrtb2.Imp{imp}
 
-		tagID, err = jsonparser.GetString(reqCopy.Imp[0].Ext, "bidder", "TagID")
-		if err != nil {
-			errs = append(errs, err)
-			continue
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, append(errs, err)
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &collossusExt); err != nil {
+			return nil, append(errs, err)
 		}
```

Comment:
I'm curious, why the change from using `jsonparser.GetString`?

Reply:
`jsonparser.GetString` returns an error if the path is missing and `json.Unmarshal` returns an empty string. I thought that focusing on the error is not entirely correct. Therefore, I replaced it. Could you share a possibly better solution? Thanks.

Reply:
I see. Thank you for explaining. I ask because `jsonparser` is orders of magnitude more efficient when accessing a single value from json. You are right that an error is returned if the path is missing, but luckily that error always has a specific value you could filter out, something like:

```
if tagID, err := jsonparser.GetString(reqCopy.Imp[0].Ext, "bidder", "TagID"); err == nil {
	reqCopy.Imp[0].TagID = tagID
} else if err !=  jsonparser.KeyPathNotFoundError {
	errs = append(errs, err)
	continue
}
```

Reply:
thanks, i like your solution! =)